naptime-graphql/src/test/scala/org/coursera/naptime/ari/graphql/helpers/ArgumentBuilder.scala

```diff
@@ -0,0 +1,43 @@
+package org.coursera.naptime.ari.graphql.helpers
+
+import org.coursera.naptime.PaginationConfiguration
+import sangria.schema.Args
+import sangria.schema.Argument
+import sangria.schema.IntType
+import sangria.schema.OptionInputType
+import sangria.schema.StringType
+
+import scala.collection.concurrent.TrieMap
+
+object ArgumentBuilder {
+  def buildArgs(argumentDefinitions: List[Argument[_]], argumentInputs: Map[String, Any]): Args = {
+    val argsWithDefault = argumentDefinitions.filter(_.defaultValue.isDefined).map(_.name).toSet
+    val optionalArgs = argumentDefinitions.filter(_.inputValueType.isOptional).map(_.name).toSet
+    val undefinedArgs = argumentDefinitions.map(_.name)
+      .filterNot(argsWithDefault.contains)
+      .filterNot(optionalArgs.contains)
+      .filterNot(argumentInputs.contains)
+      .toSet
+    val defaultInfo = argumentDefinitions
+      .filter(x => argsWithDefault.contains(x.name)).map(x => x.name -> x.defaultValue.get._1)
+    val builder = TrieMap.newBuilder[String, Any]
```

Comment:
we generally try to avoid builders and stuff for things like this because it makes it harder to see what's going on / the state of any variable at any given time. we really like immutability in scala because it makes it much easier to reason about things